packages/toolpad-app/src/toolpad/AppEditor/PageEditor/ComponentEditor.tsx

```diff
@@ -149,7 +149,7 @@ export default function ComponentEditor({ className }: ComponentEditorProps) {
 
   const { selection } = editor;
 
-  const selectedNode = selection ? appDom.getNode(dom, selection) : null;
+  const selectedNode = selection ? appDom.getMaybeNode(dom, selection) : null;
```

Comment:
This seems similar to the issue in https://github.com/mui/mui-toolpad/pull/1233.
Is there a way we could avoid using `getMaybeNode` here and keep using `getNode` somehow?
Otherwise it might be ok if there is some edge case, just making sure.

Reply:
Do you have any suggestions in mind? I've used `try {} catch` before with `getNode` but that isn't correct since this can actually be undefined 🤔

Reply:
It depends on why you made this change, which new cases were you trying to cover?
I'm taking a look to see. Ideally if there is a selected node the node should always exist.

Reply:
I guess that we maybe keep this for now and create an issue to also include the selected node in undo/redo (as not sure how simple that would be to do), and then replace these `getMaybeNode` with `getNode` again?

Reply:
I assume this happens because:
1. We add component to canvas
2. We select it
3. We undo add change
4. We still hold selected node reference hence the code fails. So `getMaybeNode` to me makes sense in this case, because the node might not exist

We could do as I mentioned in some previous comments and hold selection state inside dom, then this shouldn't be an issue

Reply:
Let's probably keep it like this for now and create an issue for figuring out how to handle the selected component when doing undo/redo, we can other see what other tools are doing.
I guess it's a complex enough problem that it should be solved separately.

Reply:
Interestingly, in figma, selecting a node seems to be considered a undoable action on its own. e.g. if I add an element and then switch selection 5 times between two elements, I need to undo 5 times before the last added element is removed.

Reply:
Ah that makes sense, I think we can do that then and store the selection in the DOM?

Reply:
I'd say not only figma, I've seen quite a few apps behaving like that. IMO it's quite useful, at least for me working with those apps